src/agentscope/agents/agent.py

```diff
@@ -72,10 +75,21 @@ def __init__(
         else:
             self.memory = None
 
+        if agent_id is None:
+            self.agent_id = self.__class__.generate_agent_id()
+        else:
+            self.agent_id = agent_id
+
         # The audience of this agent, which means if this agent generates a
         # response, it will be passed to all agents in the audience.
         self._audience = None
 
+    @classmethod
+    def generate_agent_id(cls) -> str:
+        """Generate the agent_id of this agent instance"""
+        # TODO: change cls.__name__ into a global unique agent_type
+        return f"{cls.__name__}_{uuid.uuid4().hex}"
```

Comment:
Is the id too long here?

Reply:
The agent_id is used to ensure global uniqueness. I tend to save the full agent_id here and truncate the first x digits when needed for display.